### ISSUE TITLE
Add an option to disable deprecation warnings

### ISSUE THREAD

**Issue by K0-RR** [Prospector/ModMenu#231]
```
[12:34:25] [main/WARN]: WARNING! Mod fermion is only using deprecated 'modmenu:clientsideOnly' custom value! This is no longer needed and will be removed in 1.18 snapshots.
[12:34:25] [main/WARN]: WARNING! Mod fermion is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod satin is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cloth-durability-bar-api-v1 is only using deprecated 'modmenu:parent' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cloth-durability-bar-api-v1 is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod fermion-modkeys is only using deprecated 'modmenu:parent' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod fermion-modkeys is only using deprecated 'modmenu:clientsideOnly' custom value! This is no longer needed and will be removed in 1.18 snapshots.
[12:34:25] [main/WARN]: WARNING! Mod fermion-modkeys is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod fiber is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cardinal-components-level is only using deprecated 'modmenu:parent' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cardinal-components-level is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod astromine-foundations is only using deprecated 'modmenu:parent' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod libblockattributes_items is only using deprecated 'modmenu:parent' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod libblockattributes_items is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod wrenchable is only using deprecated 'modmenu:clientsideOnly' custom value! This is no longer needed and will be removed in 1.18 snapshots.
[12:34:25] [main/WARN]: WARNING! Mod wrenchable is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod structure-helpers is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cloth-config2 is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cardinal-components-entity is only using deprecated 'modmenu:parent' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cardinal-components-entity is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod magna is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod libblockattributes_fluids is only using deprecated 'modmenu:parent' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod libblockattributes_fluids is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod staticdata is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cardinal-components-base is only using deprecated 'modmenu:parent' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cardinal-components-base is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod terraform-shapes-api-v1 is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cardinal-components-block is only using deprecated 'modmenu:parent' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cardinal-components-block is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cardinal-components-item is only using deprecated 'modmenu:parent' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cardinal-components-item is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cardinal-components-chunk is only using deprecated 'modmenu:parent' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cardinal-components-chunk is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cardinal-components-util is only using deprecated 'modmenu:parent' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cardinal-components-util is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod libblockattributes_core is only using deprecated 'modmenu:parent' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod libblockattributes_core is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cloth-basic-math is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod autoconfig1u is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cardinal-components-world is only using deprecated 'modmenu:parent' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cardinal-components-world is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cloth-datagen-api-v1 is only using deprecated 'modmenu:parent' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod cloth-datagen-api-v1 is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod libstructure is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod astromine-core is only using deprecated 'modmenu:parent' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[12:34:25] [main/WARN]: WARNING! Mod team_reborn_energy is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
```

This is only part of the spam I get every time I launch the game. Devs aren't interested in fixing this YET so there is nothing I can do. Please consider compensating it to something like
`[main/INFO]:  ̶W̶A̶R̶N̶I̶N̶G̶!̶ Mods [mod1; mod2; mod3; ...] are using deprecated 'modmenu:api' custom value (...)`
`[main/INFO]: Mods [mod1; mod2; mod3; ...] are using deprecated 'modmenu:parent' custom value.`
`[main/INFO]: Mods [mod1; mod2; mod3; ...] are using deprecated 'modmenu:clientsideOnly' custom value.`

I know it would probably require temp saving which mods are using obsolete values and then printing it after all mods are loaded but...

**Comment by K0-RR** [comment]
Or maybe add an option to disable logging these.

**Comment by coderbot16** [comment]
Being spammy is sort of the point, it's supposed to get people to bug the mod authors to update their stuff.

**Comment by K0-RR** [comment]
I know this and devs know that it’s not game-breaking. I tried reporting this to them and they only laugh like "I need to rush to have it done before these 1.18 snapshots" or like shedaniel said "I don’t care about this"

**Comment by haykam821** [comment]
I would prefer it if Mod Menu only warned when a mod sets the `modmenu:clientsideOnly` custom value without also setting the `environment` to `client` as well.

**Comment by jackassmc** [comment]
Since 1.18 snapshots have finally arrived and Mod Menu for 1.18 snapshots no longer support the legacy API and legacy custom values this has been implemented for 1.16 and 1.17. The default behavior is to log deprecation warnings for now, but this will be changed for the very last 1.16 and 1.17 Mod Menu versions.

> I would prefer it if Mod Menu only warned when a mod sets the `modmenu:clientsideOnly` custom value without also setting the `environment` to `client` as well.

This was the intended behavior, in line with how the other warnings work. This oversight has been fixed in the latest 1.16 and 1.17 Mod Menu versions.
